bigquery_etl/cli/check.py

```diff
@@ -0,0 +1,199 @@
+            )
+
+        if not detected_partition:
+            print(
+                "WARNING: No date specified for partition, attempting to check entire table"
+            )
+    return result
+
+
+def _parse_check_output(output: str) -> str:
+    output = output.replace("\n", " ")
+    if "ETL Data Check Failed:" in output:
+        return f"ETL Data Check Failed:{output.split('ETL Data Check Failed:')[1]}"
+    return output
+
+
+@click.group(
+    help="""
+        Commands for managing data checks.
+        \b
+
+        UNDER ACTIVE DEVELOPMENT See https://mozilla-hub.atlassian.net/browse/DENG-919
+        """
+)
+@click.pass_context
+def check(ctx):
+    """Create the CLI group for the check command."""
+    # create temporary directory generated content is written to
+    # the directory will be deleted automatically after the command exits
+    ctx.ensure_object(dict)
+    ctx.obj["TMP_DIR"] = ctx.with_resource(tempfile.TemporaryDirectory())
+
+
+@check.command(
+    help="""
+    Run ETL checks.
+s    \b
+
+    Example:
+     ./bqetl check run ga_derived.downloads_with_attribution_v2 --parameter download_date::2023-05-01
+    """,
+    context_settings=dict(
+        ignore_unknown_options=True,
+        allow_extra_args=True,
+    ),
+)
+@click.argument("name")
+@sql_dir_option
+@click.option(
+    "--parameter",
+    "-p",
+    help="Parameters required for processing including partition. If table is partitioned then the "
+    " format <column-name>::<yyy-mm-dd>, must be provided if not executing in Airflow. "
+    " Since some tables are not partitioned parameters are not required. ",
+    type=click.UNPROCESSED,
+    callback=_build_parameters,
+    multiple=True,
+    required=False,
+)
+@click.pass_context
+def run(ctx, name, sql_dir, parameter):
+    """Run a check."""
+    if not is_authenticated():
+        click.echo(
+            "Authentication to GCP required. Run `gcloud auth login` "
+            "and check that the project is set correctly."
+        )
+        sys.exit(1)
+
+    checks_file, project_id, dataset_id, table = paths_matching_checks_pattern(
+        name, sql_dir, project_id=None
+    )
+
+    _run_check(
+        checks_file,
+        project_id,
+        dataset_id,
+        table,
+        parameter,
+        ctx.args,
+    )
+
+
+def _run_check(
+    checks_file,
+    project_id,
+    dataset_id,
+    table,
+    parameters,
+    query_arguments,
+):
+    """Run the check."""
+    query_arguments.append("--use_legacy_sql=false")
+    if project_id is not None:
+        query_arguments.append(f"--project_id={project_id}")
+
+    if parameters is None:
+        # Convert all the Airflow params to jinja usable dict.
+        parameters = _parse_query_arguments(query_arguments)
+    else:
+        # Convert all CLI parameters to BQ query args.
+        query_arguments += _build_query_arguments(parameters)
```

Comment:
I'm wondering if parsing these parameters is actually necessary. For our scheduling configuration it is already possible to specify parameters as strings, and they just get used 1:1: https://github.com/mozilla/bigquery-etl/blob/d8f1dbf1ff56698a25e422ab60797e88c25cc875/sql/moz-fx-data-marketing-prod/ga_derived/www_site_empty_check_v1/metadata.yaml#L13

Could we do the same here? `--parameter` would just take a string and pass it into `bq query` without any parsing and or validation? If something is incorrect then it would get raised when running the check.

I'd imagine that once these checks get scheduled in Airflow we can either reuse the `parameters` under `scheduling` or add a new `checks` section in `metadata.yaml` where parameters can be specified that are used for these checks.